src/viewer.js

```diff
@@ -861,6 +861,23 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
     },
 
     /**
+     * Turns debugging mode on or off for this viewer.
+     *
+     * @function
+     * @param {Boolean} true to turn debug on, false to turn debug off.
+     * @return {OpenSeadragon.Viewer} Chainable.
+     */
+    setDebugMode: function(debugMode){
+
```

Comment:
In addition to what this function is doing, it should also set the viewer's debugMode flag as well; that'll cover future images that are added.